presentation/components/src/presentation-components/table/DataProvider.ts

```diff
@@ -256,9 +257,57 @@ const createColumns = (descriptor: Readonly<Descriptor> | undefined): ColumnDesc
     return [createLabelColumn()];
 
   const sortedFields = [...descriptor.fields].sort(priorityAndNameSortFunction);
+
+  extractSameInstanceFields(sortedFields);
+
   return sortedFields.map((field) => createColumn(field));
 };
 
+const extractSameInstanceFields = (fields: Field[]) => {

```

Comment:
It seems the function is used for two different purposes:
- replace same instance nested content fields with their nested fields (for `createColumns`)
- find same instance nested content fields (for `createRow` call)

Common pieces should be reused, but we should not be doing unnecessary work. E.g. we don't need to be creating `foundFields` map when calling it from `createColumns` and we don't need to do replacements when calling from `createRow`.

Reply:
Updated to have two different functions for each task